src/main/java/duke/Command.java

```diff
@@ -0,0 +1,25 @@
+package duke;
+
+/**
+ * Represents the commands that Duke can accept to do various tasks
+ */
+public abstract class Command {
```

Comment:
Abstracting out the Command class reduces a lot of if else's in the Parser class and makes it really clean. Nice!